docs/001_develop/03_client-capabilities/023_custom-components/03_state-management

```diff
@@ -0,0 +1,342 @@
+
+While it's important to distinguish between an attribute and a property, it's just as important to know the following about properties:
+
+- they can be defined internally in the component class without linking them to HTML attributes. These internal properties are not exposed in the DOM as attributes.
+
+> clickCount is an internal property and it’s only accessible within the component and is not reflected in the HTML markup. Only label is accessible as an HTML attribute.
+
+```shell
+import { attr, customElement, GenesisElement, html } from '@genesislcap/web-core';
+
+export class MyButton extends GenesisElemen {
+    @attr label = "Click Me"; // Linked attribute for label
+    clickCount = 0; // Internal property, not linked to an attribute
+
+    handleClick() {
+        this.clickCount++; // Update internal state
+        console.log(`Button clicked ${this.clickCount} times`);
+    }
+}
+```
+
+- They can be declared using @attr which provides reflection between the JavaScript property and the HTML attribute. Changes to the attribute update the property, and vice versa.
+
+```shell
+export class MyButton extends FASTElement {
+    @attr label = "Click Me"; // Declared as a linked property
+}
+
+```
+
+- The : symbol is used for property binding, allowing you to bind values directly to DOM properties instead of HTML attributes. This is useful when you want to bind complex data types or frequently changing properties that shouldn’t be reflected as attributes in the DOM.
+
+See `clickCount`
+
+```shell
+
+@customElement({
+  name: 'my-button',
+  template: html`
+    <<button> :clickCount="${x => x.clickCount}"></button>
+`;,
+  styles,
+})
+
+export class MyButton extends GenesisElemen {
+    @attr label = "Click Me"; // Linked attribute for label
+    clickCount = 0; // Internal property, not linked to an attribute
+
+    handleClick() {
+        this.clickCount++; // Update internal state
+        console.log(`Button clicked ${this.clickCount} times`);
+    }
+}
+
+```
+
+<!-- JavaScript values with type safety
+Reflect to attributes automatically
+Used for external API -->
+
+## Observables
+
+Observables are a powerful tool for creating reactive components. They allow properties to trigger reactivity within a component whenever the data changes, which is essential for dynamic and interactive elements. Observables are declared with the `@observable`.
+
+- Observable properties automatically trigger updates to any part of the template or component that depends on them. Whenever an observable property changes, the UI or other bound elements update without additional code.
+- Use `@observable` to mark a property as observable. This ensures that changes to the property automatically trigger any relevant updates in the component.
+
+```shell
```

Comment:
I am a bit confused by this example - we talk about observable but then don't really show it in the example. Also we're using `clickCount` on the template, but have only defined `value` on the class? Perhaps you deleted the wrong variable on the class after copy and pasting?